### PR TITLE
Fixes endless loop of HorizontalScroll

### LINKED ISSUE #20684: [Android][MAUI-8.0.7] CPU usage consistently remains above 10%, even without any operation
### Description

When there is a 'CollectionView' with property set to `ItemsLayout="HorizontalList"`, the interface responds very slowly.CPU is always occupied, maintaining around 20%

### Steps to Reproduce

Download https://github.com/BaY1251/MauiDebug.git
1. Open /HorizontalList/HorizontalList.sln 
2. Change configuration to 'Release', and deployment the project
3. Wait for 5 minutes, do not take any action during this period
4. Command line input `adb shell top -n 1 | findstr "\<com.companyname.horizontallist\> \<PID\>"'` to view CPU usage. 
5. Waiting for what you think is a long enough time
6. Command line input `adb shell top -n 1 | findstr "\<com.companyname.horizontallist\> \<PID\>"'` to view CPU usage. 

### Link to public reproduction project repository

https://github.com/BaY1251/MauiDebug.git

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.6 SR1

### Affected platforms

Android

### Affected platform versions

Android7.1

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by BaY1251**
### Description of Change

I know PR #22347 has fixed #20684. But,I am still searching for the root cause of the problem.
I printed the log and saw that the height was constantly being changed to completely different values.
I think MauiScrollView.OnLayout is where the modifications should be made.

Change \src\Core\src\Platform\Android\MauiScrollView.cs

Log:
https://github.com/dotnet/maui/pull/22566#issuecomment-2128358252

### Issues Fixed

Fixes #20684

### CODE REVIEW

File: src/Core/src/Platform/Android/MauiScrollView.cs [hunk 1]
```diff
@@ -214,7 +214,7 @@ protected override void OnLayout(bool changed, int left, int top, int right, int
 		{
 			base.OnLayout(changed, left, top, right, bottom);
 
-			if (_hScrollView?.Parent == this && _content is not null)
+			if (changed && _hScrollView?.Parent == this && _content is not null)
```
Comment (jsuarezruiz) [inline src/Core/src/Platform/Android/MauiScrollView.cs:217]: The issue still happens after merge https://github.com/dotnet/maui/pull/22347?

Comment (BaY1251) [reply src/Core/src/Platform/Android/MauiScrollView.cs:217]: PR22347  has fixed this issue and performed very well.
This PR attempts to solve the problem from another perspective, that is, the _hScrollView should not be adjusted when 'changed' is' false '.

**Comment by BaY1251** [comment]
Fixed by PR https://github.com/dotnet/maui/pull/22347

**Pull request closed without merging**